02_knight_and_princess/robert_gaisin/main.cpp

```diff
@@ -0,0 +1,235 @@
+  returned_graph.add_edge_in_vertex(0, 2, 6);
+  returned_graph.add_edge_in_vertex(0, 2, 7);
+  returned_graph.add_edge_in_vertex(0, 3, 2);
+  returned_graph.add_edge_in_vertex(0, 3, 8);
+  returned_graph.add_edge_in_vertex(0, 4, 3);
+  returned_graph.add_edge_in_vertex(0, 4, 9);
+  returned_graph.add_edge_in_vertex(0, 5, 4);
+  returned_graph.add_edge_in_vertex(0, 5, 10);
+  returned_graph.add_edge_in_vertex(0, 6, 5);
+  returned_graph.add_edge_in_vertex(0, 6, 11);
+  returned_graph.add_edge_in_vertex(0, 7, 6);
+  returned_graph.add_edge_in_vertex(0, 7, 12);
+  returned_graph.add_edge_in_vertex(0, 8, 7);
+  returned_graph.add_edge_in_vertex(0, 8, 13);
+  returned_graph.add_edge_in_vertex(0, 9, 8);
+  returned_graph.add_edge_in_vertex(0, 9, 14);
+  returned_graph.add_edge_in_vertex(0, 10, 9);
+  returned_graph.add_edge_in_vertex(0, 10, 10);
+  returned_graph.add_edge_in_vertex(0, 10, 11);
+  returned_graph.add_edge_in_vertex(0, 10, 15);
+  returned_graph.add_edge_in_vertex(0, 11, 12);
+  returned_graph.add_edge_in_vertex(0, 11, 13);
+  returned_graph.add_edge_in_vertex(0, 11, 16);
+  returned_graph.add_edge_in_vertex(0, 12, 14);
+  returned_graph.add_edge_in_vertex(0, 12, 17);
+  returned_graph.add_edge_in_vertex(0, 13, 15);
+  returned_graph.add_edge_in_vertex(0, 13, 16);
+  returned_graph.add_edge_in_vertex(0, 13, 17);
+  Edge tmp = {returned_graph.next_edge_id(), 0, 1};
+  returned_graph.add_edge(tmp);
+  Edge tmp1 = {returned_graph.next_edge_id(), 0, 2};
+  returned_graph.add_edge(tmp1);
+  Edge tmp100 = {returned_graph.next_edge_id(), 0, 3};
+  returned_graph.add_edge(tmp100);
+  Edge tmp2 = {returned_graph.next_edge_id(), 1, 4};
+  returned_graph.add_edge(tmp2);
+  Edge tmp3 = {returned_graph.next_edge_id(), 1, 5};
+  returned_graph.add_edge(tmp3);
+  Edge tmp4 = {returned_graph.next_edge_id(), 1, 6};
+  returned_graph.add_edge(tmp4);
+  Edge tmp5 = {returned_graph.next_edge_id(), 2, 7};
+  returned_graph.add_edge(tmp5);
+  Edge tmp6 = {returned_graph.next_edge_id(), 2, 8};
+  returned_graph.add_edge(tmp6);
+  Edge tmp7 = {returned_graph.next_edge_id(), 3, 9};
+  returned_graph.add_edge(tmp7);
+  Edge tmp8 = {returned_graph.next_edge_id(), 4, 10};
+  returned_graph.add_edge(tmp8);
+  Edge tmp9 = {returned_graph.next_edge_id(), 5, 10};
+  returned_graph.add_edge(tmp9);
+  Edge tmp10 = {returned_graph.next_edge_id(), 6, 10};
+  returned_graph.add_edge(tmp10);
+  Edge tmp11 = {returned_graph.next_edge_id(), 7, 11};
+  returned_graph.add_edge(tmp11);
+  Edge tmp12 = {returned_graph.next_edge_id(), 8, 11};
+  returned_graph.add_edge(tmp12);
+  Edge tmp13 = {returned_graph.next_edge_id(), 9, 12};
+  returned_graph.add_edge(tmp13);
+  Edge tmp14 = {returned_graph.next_edge_id(), 10, 13};
+  returned_graph.add_edge(tmp14);
+  Edge tmp15 = {returned_graph.next_edge_id(), 11, 13};
+  returned_graph.add_edge(tmp15);
+  Edge tmp16 = {returned_graph.next_edge_id(), 12, 13};
+  returned_graph.add_edge(tmp16);
```

Comment:
Чего-то ты тут намудрил... Сколько граней тебе нужно? 17? Значи и добавлений граней тоже должно быть 17:
```suggestion
  returned_graph.add_edge(0, 1); // от  вершины 0 до 1
  returned_graph.add_edge(0, 2); // от  вершины 0 до 2
  returned_graph.add_edge(0, 3); // от  вершины 0 до 3
  // и так далее
```